src/connectors/amazon.js

```diff
@@ -6,9 +6,17 @@ Connector.getArtist = () => {
 	return $('.trackInfoContainer .trackArtist a, .trackInfoContainer .trackArtist span').attr('title');
 };
 
+Connector.getAlbumArtist = Connector.getArtist;
+
 Connector.trackSelector = '.trackInfoContainer .trackTitle';
 
+Connector.albumSelector = '.trackSourceLink a';
+
 Connector.getAlbum = () => {
+	if ($('.trackSourceLink a').attr('href').includes('albums')) {
+		return $('.trackSourceLink a').text();
+	}
+
```

Comment:
Album tag was broken. This check fixes that.

Reply:
Does this change fix #1677?